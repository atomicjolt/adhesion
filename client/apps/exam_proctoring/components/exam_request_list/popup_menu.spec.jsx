import React from 'react';
import { shallow } from 'enzyme';
import PopupMenu from './popup_menu';

describe('popup menu', () => {
  let result;
  let props;
  let examHasStarted;
  let examHasFinished;
  let examModal;
  let message;
  beforeEach(() => {

    examHasStarted = false;
    examHasFinished = false;

    props = {
      status: 'scheduled',
      studentHasExamStarted: false,
      startExam: () => { examHasStarted = true; },
      finishExam: () => { examHasFinished = true; },
      openExamModal: () => { examModal = 1; },
      openMessageModal: () => { message = "If you're reading this, I'm too late"; },
    };

    result = shallow(<PopupMenu {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
