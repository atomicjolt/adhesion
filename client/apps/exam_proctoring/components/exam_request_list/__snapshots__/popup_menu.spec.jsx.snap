// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup menu matches the snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "boxShadow": "0px 0px 5px #555555",
      "width": "200px",
    }
  }
>
  <div
    style={
      Object {
        "borderBottom": "1px solid #ededed",
      }
    }
  >
    <AdminHoverButton
      className="qa-menu-start-btn"
      hoveredStyle={
        Object {
          "backgroundColor": "#f5f5f5",
        }
      }
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "none",
          "cursor": "pointer",
          "fontSize": "inherit",
          "padding": "20px",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      Start
    </AdminHoverButton>
  </div>
  <div
    style={
      Object {
        "borderBottom": "1px solid #ededed",
      }
    }
  >
    <AdminHoverButton
      className="qa-menu-print-btn"
      hoveredStyle={
        Object {
          "backgroundColor": "#f5f5f5",
        }
      }
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "none",
          "cursor": "pointer",
          "fontSize": "inherit",
          "padding": "20px",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      Print Test
    </AdminHoverButton>
  </div>
  <div
    style={
      Object {
        "borderBottom": "1px solid #ededed",
      }
    }
  >
    <AdminHoverButton
      className="qa-enter-answers-btn"
      hoveredStyle={
        Object {
          "backgroundColor": "#f5f5f5",
        }
      }
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "none",
          "cursor": "pointer",
          "fontSize": "inherit",
          "padding": "20px",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      Enter Answers
    </AdminHoverButton>
  </div>
  <div
    style={
      Object {
        "borderBottom": "1px solid #ededed",
      }
    }
  >
    <AdminHoverButton
      className="qa-menu-message-btn"
      hoveredStyle={
        Object {
          "backgroundColor": "#f5f5f5",
        }
      }
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "none",
          "cursor": "pointer",
          "fontSize": "inherit",
          "padding": "20px",
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      Message Student
    </AdminHoverButton>
  </div>
</div>
`;
